src/Message/PurchaseRequest.php

```diff
@@ -112,4 +116,160 @@ private function resolvePageAction()
+     * @return PurchaseRequest
+     * @throws InvalidRequestException
+     */
+    public function setPaymentConfig($type, $values = [])
+    {
+        if ($type === 'SINGLE') {
+            return $this->setParameter('paymentConfig', $type);
+        }
+
+        if (!in_array($type, ['MULTI', 'MULTI_EXT'])) {
+            throw new InvalidRequestException('type is not allowed');
+        }
+
+        if (0 !== $this->getCaptureDelay()) {
+            throw new InvalidRequestException(sprintf('Capture delay must be equals 0 when using payment config %s', $type));
+        }
+
+        if ($type === 'MULTI') {
+            $values = $this->checkAndFormatMultiPaymentConfig($values);
+        } else {
+            $values = $this->checkAndFormatMultiExtPaymentConfig($values);
+        }
+
+        $value = $type . ':' . http_build_query($values, '', ';');
+        return $this->setParameter('paymentConfig', $value);
+    }
+
+    /**
+     * @param array $settings
+     * @return array
+     * @throws InvalidRequestException
+     */
+    private function checkAndFormatMultiPaymentConfig(array $settings)
+    {
+        if (!isset($settings['first'], $settings['count'], $settings['period'])) {
+            throw new InvalidRequestException('Invalid parameter for type MULTI');
+        }
+
+        if (!is_int($settings['count']) || $settings['count'] < 1) {
+            throw new InvalidRequestException(sprintf('Invalid count %s. Expecting an integer greater than 1', $settings['count']));
+        }
+
+        if (!is_int($settings['period']) || $settings['period'] < 0) {
+            throw new InvalidRequestException(sprintf('Invalid period %s. Expecting an integer greater than 0', $settings['period']));
+        }
+
+        $settings['first'] = $this->formatPaymentConfigAmount($settings['first']);
+
+        return $settings;
+    }
+
+    /**
+     * @param array $settings
+     * @return array
+     * @throws Exception
+     * @throws InvalidRequestException
+     */
+    private function checkAndFormatMultiExtPaymentConfig(array $settings)
+    {
+        // First date must be greater or equal today
+        $minPeriodDate = (new DateTime())->format('Ymd');
+        // Last date must be less than in 1 year
+        $maxPeriodDate = (new DateTime())->modify('+1 year')->format('Ymd');
+
+        $totalAmount = 0;
+        foreach ($settings as $date => $amount) {
+            if (!Validation::validateDate($date, 'Ymd')) {
+                throw new InvalidRequestException(sprintf('Invalid date %s. Expecting format Ymd', $date));
+            }
+
+            if (!Validation::isDateInPeriod($date, $minPeriodDate, $maxPeriodDate)) {
+                throw new InvalidRequestException(sprintf('Invalid date %s. The date must be between today and in one year', $date));
+            }
+
+            $amount = $this->formatPaymentConfigAmount($amount);
+            $settings[$date] = $amount;
+            $totalAmount += (int) $amount;
```

Comment:
we shouldn't cast, no? By doing so, we could false the result

Reply:
We send price like 10.00. The number of decimals is 2 max.
Then, all amounts are formatted. It becomes 1000.